roles/translator/src/lib/sv2/upstream/message_handler.rs

```
@@ -1,40 +1,54 @@
-use crate::sv2::upstream::data::UpstreamData;
+use crate::sv2::Upstream;
 use roles_logic_sv2::{
     common_messages_sv2::{
         ChannelEndpointChanged, Reconnect, SetupConnectionError, SetupConnectionSuccess,
     },
-    handlers::common::{ParseCommonMessagesFromUpstream, SendTo as SendToCommon},
-    Error,
+    handlers_sv2::{ParseCommonMessagesFromUpstreamAsync, HandlerError},
 };
-use tracing::info;
+use tracing::{info, error};
 
-impl ParseCommonMessagesFromUpstream for UpstreamData {
-    fn handle_setup_connection_success(
-        &mut self,
-        m: SetupConnectionSuccess,
-    ) -> Result<SendToCommon, Error> {
-        info!(
-            "Received `SetupConnectionSuccess`: version={}, flags={:b}",
-            m.used_version, m.flags
-        );
-        Ok(SendToCommon::None(None))
+impl ParseCommonMessagesFromUpstreamAsync for Upstream {
+    async fn handle_setup_connection_error(
+            &mut self,
+            msg: SetupConnectionError<'_>,
+        ) -> Result<(), HandlerError> {
+            error!(
+                "Received `SetupConnectionError`: version={}, flags={:b}",
+                msg.error_code, msg.flags
+            );
+
+            Ok(())
```

Shouldn't we return an error here?

---

I was thinking more on lines liek the user can now report this directly to status handler, respawn the service.

---

Yeah that's also kind of what I was thinking about.. would this be done directly here in the handler, right? 

Or are we missing anything needed by the status handler?

---

It can be done.